src/autorun.ts

```diff
@@ -72,6 +78,14 @@ export const run = <T>(fn: Cb<T>): Observable<T> => new Observable(observer => {
         } finally {
             context.$ = prev$;
             context._ = prev_;
+            // Remove all unused deps
```

Comment:
**major:** I think, there's a bug with removing all unused deps in `finally` because of mid-flight interruption, let me describe a case and you'll judge if it's valid:

```ts
let a = interval(1000).startWith(0); // sync value
let b = of('Hello').delay(100); // async value
let c = ajax('…'); // some expensive & effect-full subscription

run(() => {
  let b = ($(a) % 2) ? $(b) : 'World'; // b is under condition
  return b + $(c); // a and c are always used
});
```

Though because of mid-flight interruption on reading `b` each other frame — `c` would be marked as "untracked" and "unused". Therefore its subscription would be dropped.

If it is correct, then we might need some transaction-like behavior. Maybe a temporary state that would be committed to the main state only if theres no mid-flight interruption because of late subscription.

These async things are hard to think about, so please double-check my arguments.

Reply:
Wow good catch!
I'll try make a test out of that.

What I'm currently thinking of:
Only do the unsubscribe when the expression didn't interrupt.
I also think somehow the tracking state must be reverted in that case...
I'll come up with some cases. Maybe we need to discuss some scenarios, we'll see...

Reply:
First I thought to initialize another Map to store the temp state.
But later... I guess additional `{ tempState: { used: boolean, tracked: boolean } }` on a `TrackEntry` might solve it best.
Which would be translated to real state and cleaned on successful run.
What do you think?